### PR TITLE
chore: add activation events

### PR DESCRIPTION
**Pull request by makeavish**
#### Screenshots

NA

### CODE REVIEW

**Review by SagarRajput-7 — APPROVED** [review]

File: frontend/src/container/ExportPanel/ExportPanelContainer.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ import {
 	Wrapper,
 } from './styles';
 import { filterOptions, getSelectOptions } from './utils';
+// import logEvent from 'api/common/logEvent';
```
Comment (YounixM) [inline frontend/src/container/ExportPanel/ExportPanelContainer.tsx:19]: Remove commented code

File: frontend/src/container/LogsExplorerList/index.tsx [hunk 2]
```diff
@@ -169,7 +169,9 @@ function LogsExplorerList({
 				!isFetching &&
 				logs.length === 0 &&
 				!isError &&
-				isFilterApplied && <EmptyLogsSearch />}
+				isFilterApplied && (
+					<EmptyLogsSearch dataSource={DataSource.LOGS} panelType="LIST" />
```
Comment (SagarRajput-7) [inline frontend/src/container/LogsExplorerList/index.tsx:173]: NIT: use constant `DataSource`

File: frontend/src/container/ExportPanel/ExportPanelContainer.tsx (hunk 1 above)
Comment (SagarRajput-7) [inline frontend/src/container/ExportPanel/ExportPanelContainer.tsx:19]: commented code

File: frontend/src/hooks/queryBuilder/useCreateAlerts.tsx [hunk 3]
```diff
@@ -17,7 +19,7 @@ import { Widgets } from 'types/api/dashboard/getAll';
 import { GlobalReducer } from 'types/reducer/globalTime';
 import { getGraphType } from 'utils/getGraphType';
 
-const useCreateAlerts = (widget?: Widgets): VoidFunction => {
+const useCreateAlerts = (widget?: Widgets, caller?: string): VoidFunction => {
```
Comment (YounixM) [inline frontend/src/hooks/queryBuilder/useCreateAlerts.tsx:22]: Why are we updating hooks for the sake of sending logEvents?  We should use whatever information we have and send events accordingly.

Comment (makeavish) [reply frontend/src/hooks/queryBuilder/useCreateAlerts.tsx:22]: Without this we can't differentiate on which screen the action was done. It's important to know screen of action.

File: frontend/src/container/LogsExplorerList/index.tsx (hunk 2 above)
Comment (makeavish) [reply frontend/src/container/LogsExplorerList/index.tsx:173]: Aren't we already using DataSource Constant?

**Review by YounixM — APPROVED** [review]

**Review by SagarRajput-7 — APPROVED** [review]

File: frontend/src/container/NewDashboard/ComponentsSlider/index.tsx [hunk 4]
```diff
@@ -20,6 +21,13 @@ function DashboardGraphSlider(): JSX.Element {
 	const onClickHandler = (name: PANEL_TYPES) => (): void => {
 		const id = uuid();
 		handleToggleDashboardSlider(false);
+		logEvent('Dashboard Detail: New panel type selected', {
+			// dashboardId: '',
```
Comment (YounixM) [inline frontend/src/container/NewDashboard/ComponentsSlider/index.tsx:25]: Remove commented code.

File: frontend/src/container/ServiceApplication/ServiceTraces/index.tsx [hunk 5]
```diff
@@ -45,6 +49,28 @@ function ServiceTraces(): JSX.Element {
 		setSkipOnboarding(true);
 	};
 
+	const logEventCalledRef = useRef(false);
+	useEffect(() => {
+		if (!logEventCalledRef.current && !isUndefined(data)) {
+			const selectedEnvironment = queries.map((val) => {
```
Comment (YounixM) [inline frontend/src/container/ServiceApplication/ServiceTraces/index.tsx:55]: Fix this logic

File: frontend/src/container/MetricsApplication/Tabs/DBCall.tsx [hunk 6]
```diff
@@ -97,6 +98,24 @@ function DBCall(): JSX.Element {
 		[servicename, tagFilterItems],
 	);
 
+	const logEventCalledRef = useRef(false);
+
+	useEffect(() => {
+		if (!logEventCalledRef.current) {
+			const selectedEnvironment = queries.find(
```
Comment (YounixM) [inline frontend/src/container/MetricsApplication/Tabs/DBCall.tsx:105]: What would the value of selectedEnvironment after this change ?

File: frontend/src/container/MetricsApplication/Tabs/DBCall.tsx [hunk 7]
```diff
@@ -97,6 +98,24 @@ function DBCall(): JSX.Element {
...
+				(val) => val.tagKey === 'resource_deployment_environment',
+			)?.tagValue;
+
+			logEvent('APM: Service detail page visited', {
+				selectedEnvironment: selectedEnvironment?.[0]?.[0],
```
Comment (YounixM) [inline frontend/src/container/MetricsApplication/Tabs/DBCall.tsx:110]: We are consuming it as array.

**Review by YounixM — APPROVED** [review]

**Review by SagarRajput-7 — APPROVED** [review]

**Review by srikanthccv — APPROVED** [review]

**Pull request merged**